### PR TITLE
[Merged by Bors] - feat(Mathlib/Analysis/SpecificLimits/Basic.lean): add iff versions of lemmas about x^n going  to 0 iff x < 1

### PR DESCRIPTION
**Pull request by faenuccio**
Add the two iff versions (for NNReals and ENNReals) of the statement that $x^n$ tends to $0$ if and only if $x<1$.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecificLimits/Basic.lean [hunk 1]
```diff
@@ -224,6 +230,23 @@ theorem ENNReal.tendsto_pow_atTop_nhds_zero_of_lt_one {r : ℝ≥0∞} (hr : r <
 @[deprecated] alias ENNReal.tendsto_pow_atTop_nhds_0_of_lt_1 :=
   ENNReal.tendsto_pow_atTop_nhds_zero_of_lt_one
 
+example (f : ℕ → ℝ≥0∞) (hf : ∀ᶠ x in atTop, f x = 1) : Tendsto f atTop (𝓝 1) := by
+  exact EventuallyEq.tendsto hf
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Basic.lean:234]: What is the purpose of this `example`?

Comment (faenuccio) [reply Mathlib/Analysis/SpecificLimits/Basic.lean:234]: To be removed 😄 I will do it straight away.

File: Mathlib/Analysis/SpecificLimits/Basic.lean [hunk 2]
```diff
@@ -224,6 +230,23 @@ theorem ENNReal.tendsto_pow_atTop_nhds_zero_of_lt_one {r : ℝ≥0∞} (hr : r <
...
+  refine ⟨fun h => ?_, ENNReal.tendsto_pow_atTop_nhds_zero_of_lt_one⟩
+  have hr : r ≠ ⊤ := by
+    have : ∀ᶠ n in atTop, (⊤ : ℝ≥0∞)^n = ⊤ := by
+      simp only [pow_eq_top_iff, ne_eq, true_and, eventually_atTop, ge_iff_le]
+      exact ⟨1, fun _ _ => by linarith⟩
+    exact fun hr => ENNReal.top_ne_zero (tendsto_nhds_unique (EventuallyEq.tendsto this) (hr ▸ h))
+  rw [← ENNReal.coe_zero, ← ENNReal.coe_toNNReal hr] at h
+  simp_rw [← ENNReal.coe_pow, tendsto_coe, NNReal.tendsto_pow_atTop_nhds_zero_iff,
+    ← ENNReal.coe_lt_coe, ENNReal.coe_toNNReal hr] at h
+  exact h
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Basic.lean:245]:
```suggestion
  refine ⟨fun h ↦ ?_, tendsto_pow_atTop_nhds_zero_of_lt_one⟩
  lift r to NNReal
  · refine fun hr ↦ top_ne_zero (tendsto_nhds_unique (EventuallyEq.tendsto ?_) (hr ▸ h))
    exact eventually_atTop.mpr ⟨1, fun _ hn ↦ pow_eq_top_iff.mpr ⟨rfl, Nat.pos_iff_ne_zero.mp hn⟩⟩
  rw [← coe_zero] at h
  norm_cast at h ⊢
  exact NNReal.tendsto_pow_atTop_nhds_zero_iff.mp h
```

**Review by fpvandoorn — COMMENTED** [review]
To avoid clashing with the name in the root namespace, let's protect these.

bors d+

File: Mathlib/Analysis/SpecificLimits/Basic.lean (hunk 2 above)
Comment (faenuccio) [reply Mathlib/Analysis/SpecificLimits/Basic.lean]: Nice, thanks!

**Review by MichaelStollBayreuth — APPROVED** [review]
LGTM

**Pull request closed without merging**